t/ui/27-plugin_obs_rsync_batched_status_details.t

```diff
@@ -0,0 +1,183 @@
+# Copyright (C) 2020 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+use Mojo::Base -strict;
+
+use FindBin;
+use lib "$FindBin::Bin/../lib";
+use Test::More;
+use Test::Mojo;
+use OpenQA::Test::Database;
+use OpenQA::Test::Case;
+use Mojo::File qw(tempdir path);
+use File::Copy::Recursive 'dircopy';
+
+use Mojolicious;
+use IO::Socket::INET;
+use Mojo::Server::Daemon;
+use Mojo::IOLoop::Server;
+use Mojo::IOLoop::ReadWriteProcess qw(process);
+use Mojo::IOLoop::ReadWriteProcess::Session 'session';
+use Time::HiRes 'sleep';
+
+my $test_case   = OpenQA::Test::Case->new;
+my $schema_name = OpenQA::Test::Database->generate_schema_name;
+my $schema      = $test_case->init_data(schema_name => $schema_name);
+
+use OpenQA::SeleniumTest;
+
+$SIG{INT} = sub {
+    session->clean;
+};
+
+END { session->clean }
+
+my $port = Mojo::IOLoop::Server->generate_port;
+my $host = "http://127.0.0.1:$port";
+
+sub fake_api_server {
+    my $mock = Mojolicious->new;
+    $mock->mode('test');
+    $mock->routes->get(
+        "/public/build/BatchedProj/_result" => sub {
+            shift->render(
+                status => 200,
+                text =>
+                  '<result project="BatchedProj" repository="images" arch="local" code="published" state="published">'
+            );
+        });
+    return $mock;
+}
+
+sub _port { IO::Socket::INET->new(PeerAddr => '127.0.0.1', PeerPort => shift) }
+
+my $daemon;
+my $mock            = Mojolicious->new;
+my $server_instance = process sub {
+    $daemon = Mojo::Server::Daemon->new(app => fake_api_server, listen => [$host]);
+    $daemon->run;
+    _exit(0);
+};
+
+sub start_server {
+    $server_instance->set_pipes(0)->start;
+    sleep 0.1 while !_port($port);
+    return;
+}
+
+sub stop_server {
+    # now kill the worker
+    $server_instance->stop();
+}
+
+$ENV{OPENQA_CONFIG} = my $tempdir = tempdir;
+my $home_template = path(__FILE__)->dirname->dirname->child('data', 'openqa-trigger-from-obs');
+my $home          = "$tempdir/openqa-trigger-from-obs";
+dircopy($home_template, $home);
+my $url = "http://127.0.0.1:$port/public/build/%%PROJECT/_result?package=000product";
```

Comment:
Looks like most of the test setup code is C&P from the previous test. It would make sense to extract it.

Reply:
I didn't want to introduce abstractions there because I don't see really good place for common plugin logic. But I did move `batch` and `batched` inside loop of `status_details` test, which probably looks better now

Reply:
The test itself is already under the non-plugin-specific directory `t/ui`. So making a module for common code within `t/lib/OpenQA/Test` shouldn't make things worse.

Reply:
But it makes difference when removing single file or cropping out functions from t/lib/*. Anyway, with single loop instead of 3 test files - it shouldn't matter